contracts/registry/src/lib.rs

```diff
@@ -885,4 +887,85 @@ mod tests {
         );
         assert_eq!(test_utils::get_logs(), vec![log_mint, log_renew].concat());
     }
+
+    #[test]
+    fn sbt_recover_basics() {
+        let (mut ctx, mut ctr) = setup(&issuer2(), 3 * MINT_DEPOSIT);
+        let m1_1 = mk_metadata(1, Some(START + 10));
+        let m2_1 = mk_metadata(2, Some(START + 10));
+        ctr.sbt_mint(vec![(alice(), vec![m1_1.clone()])]);
+        assert_eq!(ctr.sbt_supply_by_owner(alice(), issuer2(), None), 1);
+
+        //issue tokens by a different issuer
+        ctx.predecessor_account_id = issuer1();
+        testing_env!(ctx.clone());
+        ctr.sbt_mint(vec![(alice(), vec![m1_1.clone(), m2_1.clone()])]);
+        assert_eq!(ctr.sbt_supply_by_owner(alice(), issuer1(), None), 2);
+
+        ctr.sbt_recover(alice(), bob());
+        let recover_log = mk_log_str(
+            "recover",
+            &format!(
+                r#"{{"issuer":"{}","old_owner":"{}","new_owner":"{}"}}"#,
+                issuer1(),
+                alice(),
+                bob()
+            ),
+        );
+        assert_eq!(test_utils::get_logs().len(), 2);
+        assert_eq!(test_utils::get_logs()[1], recover_log[0]);
+        assert!(ctr.is_banned(alice()));
+        assert!(!ctr.is_banned(bob()));
+        assert_eq!(ctr.sbt_supply_by_owner(alice(), issuer1(), None), 0);
+        assert_eq!(ctr.sbt_supply_by_owner(bob(), issuer1(), None), 2);
+        assert_eq!(ctr.sbt_supply_by_owner(alice(), issuer2(), None), 1); //check if alice still holds the tokens issued by a different issuer
+        assert_eq!(
+            ctr.sbt_tokens_by_owner(bob(), Some(issuer1()), None, None),
+            vec![(
+                issuer1(),
+                vec![
+                    mk_owned_token(1, m1_1.clone()),
+                    mk_owned_token(2, m2_1.clone())
+                ]
+            ),]
+        );
```

Comment:
please add also other queries to the check... both the supply and `sbt_tokens...` and `sbt_token()`

Reply:
DONE